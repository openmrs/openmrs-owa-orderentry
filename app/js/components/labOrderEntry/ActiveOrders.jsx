import React from 'react';
import PropTypes from 'prop-types';

const ActiveOrders = ({ labOrderData, tests }) => (
  <div>
    <table className="table bordered mw-958-px">
      <thead>
        <tr>
          <th>Name</th>
          <th>tests</th>
          <th>status</th>
        </tr>
      </thead>
      <tbody>
        <tr key={labOrderData.uuid}>
          <td>{labOrderData.display}</td>
          <td>{tests.join(', ')}</td>
          <td>awaiting-results</td>
        </tr>
      </tbody>
    </table>
  </div>
);

export default ActiveOrders;

ActiveOrders.defaultProps = {
  labOrderData: {
    uuid: '',
    display: '',
  },
  tests: [''],
};

ActiveOrders.propTypes = {
  labOrderData: PropTypes.shape({
    uuid: PropTypes.string,
    display: PropTypes.string,
  }),
  tests: PropTypes.arrayOf(PropTypes.string),
};
