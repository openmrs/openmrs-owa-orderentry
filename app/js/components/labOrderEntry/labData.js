/**
 * This is a data point for the panel and test form field sets,
 * When an API endpoint is provided, it should be replaced with these.
 * * */
export const panelData = [
  {
    id: 1,
    name: 'Hemogram',
    concept: '6e5f5436-b56f-4c7d-9580-095588024d77',
    labCategory: 1,
    urgency: 'routine',
    tests: [
      {
        id: 1,
        test: 'Hemoglobin',
        urgency: 'routine',
        concept: '6e5f5436-b56f-4c7d-9580-095588024d77',
      },
      {
        id: 3,
        test: 'blood',
        urgency: 'routine',
        concept: '6e5f5436-b56f-4c7d-9580-095588024d77',
      },
      {
        id: 6,
        test: 'prothrombin',
        urgency: 'routine',
        concept: '6e5f5436-b56f-4c7d-9580-095588024d77',
      },
    ],
  },
  {
    id: 2,
    name: 'DLC',
    concept: '6e5f5436-b56f-4c7d-9580-095588024d77',
    labCategory: 1,
    urgency: 'routine',
    tests: [
      {
        id: 2,
        test: 'Hematocrit',
        urgency: 'routine',
        concept: '6e5f5436-b56f-4c7d-9580-095588024d77',
      },
      {
        id: 4,
        test: 'liver',
        urgency: 'routine',
        concept: '6e5f5436-b56f-4c7d-9580-095588024d77',
      },
      {
        id: 5,
        test: 'sickling',
        urgency: 'routine',
        concept: '6e5f5436-b56f-4c7d-9580-095588024d77',
      },
    ],
  },
];
export const testsData = [
  {
    id: 1,
    test: 'Hemoglobin',
    urgency: 'routine',
    concept: '6e5f5436-b56f-4c7d-9580-095588024d77',
  },
  {
    id: 2,
    test: 'Hematocrit',
    urgency: 'routine',
    concept: '6e5f5436-b56f-4c7d-9580-095588024d77',
  },
  {
    id: 3,
    test: 'blood',
    urgency: 'routine',
    concept: '6e5f5436-b56f-4c7d-9580-095588024d77',
  },
  {
    id: 4,
    test: 'liver',
    urgency: 'routine',
    concept: '6e5f5436-b56f-4c7d-9580-095588024d77',
  },
  {
    id: 5,
    test: 'sickling',
    urgency: 'routine',
    concept: '6e5f5436-b56f-4c7d-9580-095588024d77',
  },
  {
    id: 6,
    test: 'prothrombin',
    urgency: 'routine',
    concept: '6e5f5436-b56f-4c7d-9580-095588024d77',
  },
  {
    id: 7,
    test: 'Electrophore',
    urgency: 'routine',
    concept: '6e5f5436-b56f-4c7d-9580-095588024d77',
  },
  {
    id: 8,
    test: 'HbA1c',
    urgency: 'routine',
    concept: '6e5f5436-b56f-4c7d-9580-095588024d77',
  },
];
export const labCategories = [
  { id: 1, name: 'blood' },
  { id: 2, name: 'serum' },
  { id: 3, name: 'urine' },
  { id: 4, name: 'semen' },
  { id: 5, name: 'stool' },
  { id: 6, name: 'tissue' },
];
