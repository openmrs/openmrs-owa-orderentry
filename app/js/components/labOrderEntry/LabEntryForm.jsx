import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { Accordion } from '@openmrs/react-components';
import LabPanelFieldSet from './LabPanelFieldSet';
import LabTestFieldSet from './LabTestFieldSet';
import LabDraftOrder from './LabDraftOrder';
import ActiveOrders from './ActiveOrders';
import PastOrders from './PastOrders';
import createLabOrder from '../../actions/createLabOrder';
import { successToast, errorToast } from '../../utils/toast';
import {
  addTestPanelToDraft,
  removeTestPanelFromDraft,
  removeTestFromDraft,
  addTestToDraft,
  deleteDraftLabOrder,
  toggleDraftLabOrdersUgency,
} from '../../actions/draftLabOrderAction';
import '../../../css/grid.scss';
import './styles.scss';

import fetchLabOrders from '../../actions/labOrders/fetchLabOrders';
import fetchLabConcepts from '../../actions/labOrders/labConceptsAction';

export class LabEntryForm extends PureComponent {
  static propTypes = {
    createLabOrderReducer: PropTypes.shape({
      status: PropTypes.objectOf(PropTypes.bool),
      errorMessage: PropTypes.string,
      labOrderData: PropTypes.object,
    }),
    draftLabOrders: PropTypes.arrayOf(PropTypes.any).isRequired,
    dateFormat: PropTypes.string.isRequired,
    selectedLabPanels: PropTypes.arrayOf(PropTypes.any).isRequired,
    defaultTests: PropTypes.arrayOf(PropTypes.any).isRequired,
    selectedTests: PropTypes.arrayOf(PropTypes.any).isRequired,
    dispatch: PropTypes.func.isRequired,
    labOrders: PropTypes.shape({
      results: PropTypes.arrayOf(PropTypes.objectOf(PropTypes.any)),
    }),
    patient: PropTypes.shape({
      uuid: PropTypes.string,
    }),
    encounterType: PropTypes.shape({
      uuid: PropTypes.string,
    }).isRequired,
    encounterRole: PropTypes.shape({
      uuid: PropTypes.string,
    }),
    inpatientCareSetting: PropTypes.shape({
      uuid: PropTypes.string,
    }),
    conceptsAsPanels: PropTypes.array,
    standAloneTests: PropTypes.array,
    session: PropTypes.shape({
      currentProvider: PropTypes.shape({
        person: PropTypes.shape({
          uuid: PropTypes.string,
        }),
        uuid: PropTypes.string,
      }),
      currentLocation: PropTypes.object,
    }),
    orderables: PropTypes.arrayOf(PropTypes.object).isRequired,
    getLabOrderables: PropTypes.string.isRequired,
  };

  static defaultProps = {
    createLabOrderReducer: {
      status: {},
      errorMessage: '',
    },
    encounterRole: {
      uuid: '',
    },
    patient: {
      uuid: '',
    },
    inpatientCareSetting: {
      uuid: '',
    },
    conceptsAsPanels: [],
    standAloneTests: [],
    session: {
      currentProvider: {
        person: {
          uuid: '',
        },
      },
    },
    labOrders: {
      results: [],
    },
  };

  state = {
    categoryUUID: this.props.orderables[0].uuid || 0,
    categoryName: this.props.orderables[0].display,
    selectedPanelIds: [],
    selectedPanelTestIds: [],
  };

  componentDidMount() {
    this.props.dispatch(fetchLabOrders(null, this.props.patient.uuid));
    if (this.state.categoryUUID) {
      this.props.dispatch(fetchLabConcepts(`${this.state.categoryUUID}?v=full`));
    }
  }

  componentWillReceiveProps(nextProps) {
    const {
      selectedLabPanels,
      defaultTests,
    } = nextProps;
    this.setState({
      selectedPanelIds: selectedLabPanels.map(panel => panel.uuid),
      selectedPanelTestIds: defaultTests.map(test => test.uuid),
    });
  }

  componentDidUpdate(prevProps, prevState) {
    const {
      status: { added, error },
      errorMessage,
      labOrderData,
    } = this.props.createLabOrderReducer;
    if (added && labOrderData !== prevProps.createLabOrderReducer.labOrderData) {
      successToast('lab order successfully created');
      this.props.dispatch(fetchLabOrders(null, this.props.patient.uuid));
    }
    if (error) {
      errorToast(errorMessage);
    }
    if (this.state.categoryUUID !== prevState.categoryUUID) {
      this.props.dispatch(fetchLabConcepts(`${this.state.categoryUUID}?v=full`));
    }
  }

  handleTestSelection = (item, type) => {
    const {
      props: {
        draftLabOrders,
        dispatch,
      },
      state: {
        selectedPanelIds,
        selectedPanelTestIds,
      },
    } = this;
    const isSelected = !!draftLabOrders.filter(order => order.uuid === item.uuid).length;

    if ((type === 'panel')) {
      const isSelectedPanel = selectedPanelIds.includes(item.uuid);
      if (isSelectedPanel) {
        dispatch(removeTestPanelFromDraft(item));
      } else {
        dispatch(addTestPanelToDraft(item));
      }
    }
    if (type === 'single') {
      const isSelectedPanelTest = selectedPanelTestIds.includes(item.uuid);
      if (!isSelectedPanelTest && isSelected) dispatch(removeTestFromDraft(item));
      if (!isSelectedPanelTest && !isSelected)dispatch(addTestToDraft(item));
    }
  }

  handleUrgencyChange = (order) => {
    this.props.dispatch(toggleDraftLabOrdersUgency(order));
  }

  discardTestsInDraft = (test, action) => {
    const { dispatch } = this.props;
    if (action === 'single') return dispatch(removeTestFromDraft(test));
    if (action === 'panel') return dispatch(removeTestPanelFromDraft(test));
    return dispatch(deleteDraftLabOrder());
  }

  showFieldSet = () => (
    <div>
      <LabPanelFieldSet
        labCategoryName={this.state.categoryName}
        handleTestSelection={this.handleTestSelection}
        panels={this.props.conceptsAsPanels}
        selectedPanelIds={this.state.selectedPanelIds}
      />
      <LabTestFieldSet
        labCategoryName={this.state.categoryName}
        handleTestSelection={this.handleTestSelection}
        selectedTests={this.props.selectedTests}
        tests={this.props.standAloneTests}
      />
    </div>
  );

  changeLabForm = (id, name) => {
    this.setState({
      categoryUUID: id,
      categoryName: name,
    });
  };

  handleSubmit = () => {
    const { draftLabOrders } = this.props;
    const isEmpty = !draftLabOrders.length;
    if (isEmpty) return;
    const orders = draftLabOrders.map(labOrder => (
      {
        concept: labOrder.uuid,
        careSetting: this.props.inpatientCareSetting.uuid,
        encounter: this.props.encounterType.uuid,
        orderer: this.props.session.currentProvider.uuid,
        patient: this.props.patient.uuid,
        type: 'testorder',
        urgency: labOrder.urgency || 'ROUTINE',
      }
    ));

    const encounterPayload = {
      encounterProviders: [
        {
          encounterRole: this.props.encounterRole.uuid,
          provider: this.props.session.currentProvider.uuid,
        },
      ],
      encounterType: this.props.encounterType.uuid,
      location: this.props.session.currentLocation,
      orders,
      patient: this.props.patient.uuid,
    };
    this.props.dispatch(createLabOrder(encounterPayload));
    this.props.dispatch(deleteDraftLabOrder());
  };

  renderPendingOrders = () => {
    const { labOrderData } = this.props.createLabOrderReducer;
    if (labOrderData.orders) {
      const tests = labOrderData.orders.map(order => order.display);
      return (
        <Accordion open title="Pending Lab Orders">
          <ActiveOrders labOrderData={labOrderData} tests={tests} />
        </Accordion>
      );
    }
    return (
      <Accordion open title="Pending Lab Orders">
        <p>No pending orders</p>
      </Accordion>
    );
  };

  renderPastOrders = () => {
    const { labOrders: { results }, dateFormat } = this.props;
    if (results.length > 0) {
      return (
        <Accordion open title="Past Lab Orders">
          <PastOrders orders={results} dateFormat={dateFormat} />
        </Accordion>
      );
    }
    return (
      <Accordion title="Past Lab Orders">
        <p>No past orders</p>
      </Accordion>
    );
  };

  renderLabDraftOrder = () => (
    <div className="draft-lab-wrapper">
      <LabDraftOrder
        toggleDraftLabOrdersUgency={this.handleUrgencyChange}
        handleDraftDiscard={this.discardTestsInDraft}
        draftLabOrders={this.props.draftLabOrders}
        panelTests={this.state.selectedPanelTestIds}
        handleSubmit={() => this.handleSubmit()}
      />
    </div>
  );

  render() {
    const {
      handleCancel, handleSubmit, renderPendingOrders, renderPastOrders,
      props: { orderables, getLabOrderables },
    } = this;
    return (
      <React.Fragment>
        {
          getLabOrderables ?
            <div className="lab-order-entry">
              <h5>New Lab Order</h5>
              <br />
              <div className="lab-form-wrapper">
                <div className="lab-category">
                  <ul>
                    {orderables.map((orderable, index) => (
                      // eslint-disable-next-line
                      <li key={index}>
                        <a
                          className={this.state.categoryUUID === orderable.uuid ? 'active-category' : ''}
                          href="#"
                          id="category-button"
                          onClick={() => this.changeLabForm(orderable.uuid, orderable.display)}>
                          {orderable.display}
                        </a>
                      </li>
                    ))}
                  </ul>
                  {this.renderLabDraftOrder()}
                </div>
                <div className="order-form-wrapper">
                  <form className="lab-form simple-form-ui">{this.showFieldSet()}</form>
                </div>
              </div>
              <br />
              {renderPendingOrders()}
              <br />
              {(this.props.labOrders.results) && <div>{renderPastOrders()}</div>}
              <br />
            </div>
            :
            <p>No Lab Orderables was found</p>
        }
      </React.Fragment>
    );
  }
}

export const mapStateToProps = ({
  draftReducer: {
    draftLabOrders: {
      orders,
      selectedLabPanels,
      defaultTests,
      selectedTests,
    },
  },
  dateFormatReducer: { dateFormat },
  labConceptsReducer: {
    conceptsAsPanels,
    standAloneTests,
  },
  openmrs: { session },
  fetchLabOrderReducer: { labOrders },
  patientReducer: { patient },
  encounterRoleReducer: { encounterRole },
  encounterReducer: { encounterType },
  careSettingReducer: { inpatientCareSetting },
  createLabOrderReducer,
  labOrderableReducer: { orderables },
  getLabOrderablesReducer: { getLabOrderables },
}) => ({
  draftLabOrders: orders,
  dateFormat,
  conceptsAsPanels,
  standAloneTests,
  selectedLabPanels,
  defaultTests,
  selectedTests,
  labOrders,
  encounterType,
  inpatientCareSetting,
  createLabOrderReducer,
  encounterRole,
  patient,
  session,
  orderables,
  getLabOrderables,
});

export default connect(mapStateToProps)(LabEntryForm);
