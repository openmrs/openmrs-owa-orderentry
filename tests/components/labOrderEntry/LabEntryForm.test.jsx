import React from 'react';
import { LabEntryForm, mapStateToProps } from '../../../app/js/components/labOrderEntry/LabEntryForm';
import {
  addDraftLabOrders,
  deleteDraftLabOrder,
} from '../../../app/js/actions/draftLabOrderAction';

import { panelData, testsData } from '../../../app/js/components/labOrderEntry/labData';

let props;
let mountedComponent;

props = {
  draftLabOrders: [
    { display: 'Hemoglobin', uuid: '12746hfgjff' },
    { display: 'Hematocrit', uuid: '12746hfgjff' },
    { display: 'blood', uuid: '12746hfgjff' },
  ],
  defaultTests: [
    { display: 'Hemoglobin', uuid: '12746hfgjff' },
    { display: 'Hematocrit', uuid: '12746hfgjff' },
    { display: 'blood', uuid: '12746hfgjff' },
  ],
  dateFormat: 'DD-MM-YYYY HH:mm',
  selectedTests: [
    { display: 'Hemoglobin', uuid: '12746hfgjff' },
    { display: 'Hematocrit', uuid: '12746hfgjff' },
    { display: 'blood', uuid: '12746hfgjff' },
  ],
  selectedLabPanels: [panelData[0]],
  dispatch: jest.fn(),
  addDraftLabOrdersToStore: addDraftLabOrders,
  deleteDraftLabOrderFromStore: deleteDraftLabOrder,
  fetchLabOrders: jest.fn(),
  labOrders: {
    results: [
      {
        uuid: '1',
        dateActivated: '2018-08-27T10:23:22.228Z',
        concept: {
          display: 'Hemoglobin blood test',
        },
      },
    ],
  },
  createLabOrder: jest.fn(() => {}),
  createLabOrderReducer: {
    status: {
      error: false,
      added: true,
    },
    labOrderData: {},
  },
  session: {
    currentProvider: {
      uuid: 'jfhfhiu77474',
    },
  },
  patient: {
    uuid: 'jfgfhfgf',
  },
  conceptsAsTests: [],
  conceptsAsPanels: [],
  encounterType: {
    uuid: 'fhhfgfh9998',
  },
  inpatientCareSetting: {
    uuid: 'ifffy9847464',
  },
  encounterRole: {
    uuid: '1234trrrrr',
  },
  orderables: [
    { uuid: 'ifffy9847464', display: 'Hemoglobin', concept: '12746hfgjff' },
    { uuid: 'iewue7wyu64', display: 'Hematocrit', concept: '12746hfgjff' },
    { uuid: 'iuweihiYWGD64', display: 'blood', concept: '12746hfgjff' },
  ],
  getLabOrderables: 'buo434873euhwiqeroq',
};

let mockPanel = panelData[0];

const mockTest = testsData[0];

const getComponent = () => {
  if (!mountedComponent) {
    mountedComponent = mount(<LabEntryForm {...props} />);
  }
  return mountedComponent;
};

mockPanel = {
  uuid: '888ya-kkk',
  display: 'Concept B',
  set: true,
  setMembers: [
    { uuid: '456Abc-123', name: 'Concept D', set: false },
    { uuid: '138Abc-466', name: 'Concept E', set: false },
    { uuid: '123Def-456', name: 'Concept F', set: false },
  ],
},

describe('Component: LabEntryForm', () => {
  beforeEach(() => {
    mountedComponent = undefined;
  });

  it('should render on initial setup', () => {
    const component = getComponent();
    mapStateToProps({
      draftReducer: { draftLabOrders: { orders: [] } },
      dateFormatReducer: { dateFormat: 'DD-MM-YYYY HH:mm' },
      patientReducer: { patient: {} },
      fetchLabOrderReducer: { labOrders: [] },
      labConceptsReducer: { conceptsAsTests: [], conceptsAsPanels: [] },
      openmrs: { session: {} },
      encounterRoleReducer: { encounterRole: {} },
      encounterReducer: { encounterType: {} },
      careSettingReducer: { inpatientCareSetting: {} },
      labOrderableReducer: { orderables: [] },
      getLabOrderablesReducer: { getLabOrderables: '' },
    });
    expect(component).toMatchSnapshot();
  });

  it('should be able to select a category', () => {
    const component = getComponent();
    const categoryButton = component.find('#category-button').at(1); // click on the second category iwth id of 2
    categoryButton.simulate('click', {
      target: {},
    });
    expect(component.state().categoryUUID).toEqual('iewue7wyu64');
  });

  it('should dispatch an action to remove a test panel from the draft', () => {
    const instance = getComponent().instance();
    instance.state.selectedPanelIds = [mockPanel.uuid];

    const dipatch = jest.spyOn(props, 'dispatch');
    instance.handleTestSelection(mockPanel, 'panel');
    expect(dipatch).toBeCalled();
  });

  it('should dispatch an action to add a test panel to the draft', () => {
    const instance = getComponent().instance();

    const dipatch = jest.spyOn(props, 'dispatch');
    instance.handleTestSelection(mockPanel, 'panel');
    expect(dipatch).toBeCalled();
  });

  it('should dispatch an action to handle single test selection', () => {
    const instance = getComponent().instance();
    instance.state.selectedPanelIds = [1];

    const dispatch = jest.spyOn(props, 'dispatch');
    instance.handleTestSelection(mockTest, 'single');
    expect(dispatch).toBeCalled();
  });

  it('should dispatch an action to handle single test deletion', () => {
    const instance = getComponent().instance();
    instance.state.selectedPanelIds = [1];

    const dispatch = jest.spyOn(props, 'dispatch');
    instance.discardTestsInDraft(mockTest, 'single');
    expect(dispatch).toBeCalled();
  });

  it('should dispatch an action to handle panel deletion', () => {
    const instance = getComponent().instance();
    instance.state.selectedPanelIds = [1];

    const dispatch = jest.spyOn(props, 'dispatch');
    instance.discardTestsInDraft(mockPanel, 'panel');
    expect(dispatch).toBeCalled();
  });

  it('should dispatch an action to handle deletion of all items from the draft', () => {
    const instance = getComponent().instance();
    instance.state.selectedPanelIds = [1];

    const dispatch = jest.spyOn(props, 'dispatch');
    instance.discardTestsInDraft();
    expect(dispatch).toBeCalled();
  });

  it('should toggle the urgency state of a test', () => {
    const instance = getComponent().instance();
    const dispatch = jest.spyOn(props, 'dispatch');
    instance.handleUrgencyChange();
    expect(dispatch).toBeCalled();
  });

  it(`should change the default lab form's tests category by toggling component state`, () => {
    const component = getComponent();
    const instance = component.instance();
    const defaultCategory = instance.state.categoryUUID;
    component
      .find('#category-button')
      .at(2)
      .simulate('click', {});
    expect(instance.state.categoryUUID !== defaultCategory);
  });

  it('shows a toast prompt when test is submitted successfully', () => {
    const component = getComponent();
    const addButton = component.find('.confirm');
    addButton.simulate('click', {});

    component.setProps({
      ...component.props(),
      createLabOrderReducer: {
        status: {
          error: false,
          added: true,
        },
        labOrderData: { uuid: 'kjdhggf', display: 'order Entry', orders: [{ display: 'true' }] },
      },
    });
    expect(global.toastrMessage).toEqual('lab order successfully created');
  });

  it('shows a toast prompt when there is an error in submission', () => {
    const component = getComponent();
    const addButton = component.find('.confirm').at(0);
    addButton.simulate('click', {});
    component.setProps({
      ...component.props(),
      createLabOrderReducer: {
        status: {
          error: true,
          added: false,
        },
        labOrderData: {},
        errorMessage: 'an error occured',
      },
    });
    expect(global.toastrMessage).toEqual('an error occured');
  });

  it(`does not render the past order component if the
  length of the results props is less than 1`, () => {
    const component = getComponent();
    component.setProps({
      ...component.props(),
      labOrders: { results: [] },
    });
    expect(component.find('PastOrders').exists()).toBeFalsy();
  });

  it('should have lab orderables displayed in the category list', () => {
    const component = getComponent();
    const categoryButton = component.find('#category-button');
    expect(categoryButton.length).toEqual(3);
  });
});
