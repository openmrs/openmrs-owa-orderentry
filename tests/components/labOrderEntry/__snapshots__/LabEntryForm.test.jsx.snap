// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: LabEntryForm should render on initial setup 1`] = `
<LabEntryForm
  conceptsAsPanels={Array []}
  conceptsAsTests={Array []}
  createLabOrder={[MockFunction]}
  createLabOrderReducer={
    Object {
      "labOrderData": Object {},
      "status": Object {
        "added": true,
        "error": false,
      },
    }
  }
  dateFormat="DD-MM-YYYY HH:mm"
  defaultTests={
    Array [
      Object {
        "display": "Hemoglobin",
        "uuid": "12746hfgjff",
      },
      Object {
        "display": "Hematocrit",
        "uuid": "12746hfgjff",
      },
      Object {
        "display": "blood",
        "uuid": "12746hfgjff",
      },
    ]
  }
  deleteDraftLabOrderFromStore={[Function]}
  dispatch={
    [MockFunction] {
      "calls": Array [
        Array [
          Object {
            "payload": Promise {},
            "type": "FETCH_LAB_ORDERS",
          },
        ],
        Array [
          Object {
            "payload": Promise {},
            "type": "FETCH_LAB_CONCEPTS",
          },
        ],
      ],
    }
  }
  draftLabOrders={
    Array [
      Object {
        "display": "Hemoglobin",
        "uuid": "12746hfgjff",
      },
      Object {
        "display": "Hematocrit",
        "uuid": "12746hfgjff",
      },
      Object {
        "display": "blood",
        "uuid": "12746hfgjff",
      },
    ]
  }
  encounterRole={
    Object {
      "uuid": "1234trrrrr",
    }
  }
  encounterType={
    Object {
      "uuid": "fhhfgfh9998",
    }
  }
  fetchLabOrders={[MockFunction]}
  getLabOrderables="buo434873euhwiqeroq"
  inpatientCareSetting={
    Object {
      "uuid": "ifffy9847464",
    }
  }
  labOrders={
    Object {
      "results": Array [
        Object {
          "concept": Object {
            "display": "Hemoglobin blood test",
          },
          "dateActivated": "2018-08-27T10:23:22.228Z",
          "uuid": "1",
        },
      ],
    }
  }
  orderables={
    Array [
      Object {
        "concept": "12746hfgjff",
        "display": "Hemoglobin",
        "uuid": "ifffy9847464",
      },
      Object {
        "concept": "12746hfgjff",
        "display": "Hematocrit",
        "uuid": "iewue7wyu64",
      },
      Object {
        "concept": "12746hfgjff",
        "display": "blood",
        "uuid": "iuweihiYWGD64",
      },
    ]
  }
  patient={
    Object {
      "uuid": "jfgfhfgf",
    }
  }
  selectedLabPanels={
    Array [
      Object {
        "concept": "6e5f5436-b56f-4c7d-9580-095588024d77",
        "id": 1,
        "labCategory": 1,
        "name": "Hemogram",
        "tests": Array [
          Object {
            "concept": "6e5f5436-b56f-4c7d-9580-095588024d77",
            "id": 1,
            "test": "Hemoglobin",
            "urgency": "routine",
          },
          Object {
            "concept": "6e5f5436-b56f-4c7d-9580-095588024d77",
            "id": 3,
            "test": "blood",
            "urgency": "routine",
          },
          Object {
            "concept": "6e5f5436-b56f-4c7d-9580-095588024d77",
            "id": 6,
            "test": "prothrombin",
            "urgency": "routine",
          },
        ],
        "urgency": "routine",
      },
    ]
  }
  selectedTests={
    Array [
      Object {
        "display": "Hemoglobin",
        "uuid": "12746hfgjff",
      },
      Object {
        "display": "Hematocrit",
        "uuid": "12746hfgjff",
      },
      Object {
        "display": "blood",
        "uuid": "12746hfgjff",
      },
    ]
  }
  session={
    Object {
      "currentProvider": Object {
        "uuid": "jfhfhiu77474",
      },
    }
  }
  standAloneTests={Array []}
>
  <div
    className="lab-order-entry"
  >
    <h5>
      New Lab Order
    </h5>
    <br />
    <div
      className="lab-form-wrapper"
    >
      <div
        className="lab-category"
      >
        <ul>
          <li
            key="0"
          >
            <a
              className="active-category"
              href="#"
              id="category-button"
              onClick={[Function]}
            >
              Hemoglobin
            </a>
          </li>
          <li
            key="1"
          >
            <a
              className=""
              href="#"
              id="category-button"
              onClick={[Function]}
            >
              Hematocrit
            </a>
          </li>
          <li
            key="2"
          >
            <a
              className=""
              href="#"
              id="category-button"
              onClick={[Function]}
            >
              blood
            </a>
          </li>
        </ul>
        <div
          className="draft-lab-wrapper"
        >
          <LabDraftOrder
            draftLabOrders={
              Array [
                Object {
                  "display": "Hemoglobin",
                  "uuid": "12746hfgjff",
                },
                Object {
                  "display": "Hematocrit",
                  "uuid": "12746hfgjff",
                },
                Object {
                  "display": "blood",
                  "uuid": "12746hfgjff",
                },
              ]
            }
            handleDraftDiscard={[Function]}
            handleSubmit={[Function]}
            panelTests={Array []}
            toggleDraftLabOrdersUgency={[Function]}
          >
            <div
              className="draft-spacing draft-lab-layout"
            >
              <h5
                className="h5-draft-header"
              >
                Unsaved Draft Orders (
                3
                )
              </h5>
              <div
                className="table-container"
              >
                <ul
                  className="draft-list-container"
                >
                  <li
                    className="draft-list small-font"
                    key="0"
                  >
                    <span
                      className="draft-name"
                    >
                      hemoglobin
                    </span>
                    <div
                      className="action-btn-wrapper"
                    >
                      <span
                        className="action-btn"
                      >
                        <a
                          href="#"
                          id="draft-toggle-btn"
                          onClick={[Function]}
                        >
                          <i
                            className="icon-warning-sign scale i-gray"
                            title="Urgency"
                          />
                        </a>
                      </span>
                      <span
                        className="action-btn right"
                      >
                        <a
                          href="#"
                          id="draft-discard-btn"
                          onClick={[Function]}
                        >
                          <i
                            className="icon-remove scale"
                            title="Discard"
                          />
                        </a>
                      </span>
                    </div>
                  </li>
                  <li
                    className="draft-list small-font"
                    key="1"
                  >
                    <span
                      className="draft-name"
                    >
                      hematocrit
                    </span>
                    <div
                      className="action-btn-wrapper"
                    >
                      <span
                        className="action-btn"
                      >
                        <a
                          href="#"
                          id="draft-toggle-btn"
                          onClick={[Function]}
                        >
                          <i
                            className="icon-warning-sign scale i-gray"
                            title="Urgency"
                          />
                        </a>
                      </span>
                      <span
                        className="action-btn right"
                      >
                        <a
                          href="#"
                          id="draft-discard-btn"
                          onClick={[Function]}
                        >
                          <i
                            className="icon-remove scale"
                            title="Discard"
                          />
                        </a>
                      </span>
                    </div>
                  </li>
                  <li
                    className="draft-list small-font"
                    key="2"
                  >
                    <span
                      className="draft-name"
                    >
                      blood
                    </span>
                    <div
                      className="action-btn-wrapper"
                    >
                      <span
                        className="action-btn"
                      >
                        <a
                          href="#"
                          id="draft-toggle-btn"
                          onClick={[Function]}
                        >
                          <i
                            className="icon-warning-sign scale i-gray"
                            title="Urgency"
                          />
                        </a>
                      </span>
                      <span
                        className="action-btn right"
                      >
                        <a
                          href="#"
                          id="draft-discard-btn"
                          onClick={[Function]}
                        >
                          <i
                            className="icon-remove scale"
                            title="Discard"
                          />
                        </a>
                      </span>
                    </div>
                  </li>
                </ul>
              </div>
              <br />
              <input
                className="button cancel modified-btn"
                disabled={false}
                id="draft-discard-all"
                onClick={[Function]}
                type="button"
                value="Discard All"
              />
              <input
                className="button confirm right modified-btn"
                disabled={false}
                onClick={[Function]}
                type="submit"
                value="Sign and Save"
              />
            </div>
          </LabDraftOrder>
        </div>
      </div>
      <div
        className="order-form-wrapper"
      >
        <form
          className="lab-form simple-form-ui"
        >
          <div>
            <LabPanelFieldSet
              handleTestSelection={[Function]}
              labCategoryName="Hemoglobin"
              panels={Array []}
              selectedPanelIds={Array []}
            >
              <fieldset
                className="fieldset"
              >
                <legend>
                  Panels
                </legend>
                <div
                  className="panel-box"
                >
                  <p>
                    Hemoglobin
                     has no panels
                  </p>
                </div>
              </fieldset>
            </LabPanelFieldSet>
            <LabTestFieldSet
              handleTestSelection={[Function]}
              labCategoryName="Hemoglobin"
              selectedTests={
                Array [
                  Object {
                    "display": "Hemoglobin",
                    "uuid": "12746hfgjff",
                  },
                  Object {
                    "display": "Hematocrit",
                    "uuid": "12746hfgjff",
                  },
                  Object {
                    "display": "blood",
                    "uuid": "12746hfgjff",
                  },
                ]
              }
              tests={Array []}
            >
              <fieldset
                className="fieldset"
              >
                <legend>
                  Tests
                </legend>
                <div
                  className="tests-box"
                >
                  <p>
                    Hemoglobin
                     has no Standalone Tests
                  </p>
                </div>
              </fieldset>
            </LabTestFieldSet>
          </div>
        </form>
      </div>
    </div>
    <br />
    <Accordion
      border={false}
      open={true}
      title="Pending Lab Orders"
    >
      <div
        className="accordion "
      >
        <div
          className="header"
          onClick={[Function]}
          role="button"
          tabIndex={0}
        >
          <a>
            <span>
              <img
                className="rotate90"
                height="12px"
                src="test-file-stub"
                width="12px"
              />
            </span>
              
            Pending Lab Orders
          </a>
        </div>
        <div
          className="content open"
        >
          <p>
            No pending orders
          </p>
        </div>
      </div>
    </Accordion>
    <br />
    <div>
      <Accordion
        border={false}
        open={true}
        title="Past Lab Orders"
      >
        <div
          className="accordion "
        >
          <div
            className="header"
            onClick={[Function]}
            role="button"
            tabIndex={0}
          >
            <a>
              <span>
                <img
                  className="rotate90"
                  height="12px"
                  src="test-file-stub"
                  width="12px"
                />
              </span>
                
              Past Lab Orders
            </a>
          </div>
          <div
            className="content open"
          >
            <PastOrders
              dateFormat="DD-MM-YYYY HH:mm"
              orders={
                Array [
                  Object {
                    "concept": Object {
                      "display": "Hemoglobin blood test",
                    },
                    "dateActivated": "2018-08-27T10:23:22.228Z",
                    "uuid": "1",
                  },
                ]
              }
            >
              <div>
                <table
                  className="table bordered mw-958-px"
                >
                  <thead>
                    <tr>
                      <th
                        className="w-155-px"
                      >
                        Date
                      </th>
                      <th>
                        Details
                      </th>
                    </tr>
                  </thead>
                  <tbody>
                    <tr
                      key="1"
                    >
                      <td>
                        27-08-2018 11:23
                      </td>
                      <td>
                        Hemoglobin blood test
                      </td>
                    </tr>
                  </tbody>
                </table>
              </div>
            </PastOrders>
          </div>
        </div>
      </Accordion>
    </div>
    <br />
  </div>
</LabEntryForm>
`;
